### ISSUE TITLE
lingui extract should fail with a non-zero exit code when there are extraction failures

### ISSUE THREAD

**Issue by thekip** [lingui/js-lingui#1419]
There was already a work on that:
https://github.com/lingui/js-lingui/pull/1299

But still there are cases where errors just logged to console and zero code is returned. 

For example, "Missing Message ID" warning. 

**What should be done**: 

We should add a flag to the cli extract, for example `--strict` which will exit with non-zero code in case of any extraction problems encountered.

**Comment by landsman** [comment]
I was thinking how to hot-fix this today as well.

**Comment by thekip** [comment]
Hi, we are currently reworked an entire codebase for cli and extractor in V4. And this task already scheduled for new upcoming release. So "hofixing" to the main branch wouldn't have a lot of sense. If you want to help, you can implement this for the "next" branch

**Comment by andrii-bodnar** [comment]
@thekip is this still relevant for the v4?

**Comment by thekip** [comment]
Yes. Few of the checks are still "warning only", the big changes should be made to the codebase to stop processing after encountering these warnings.

**Comment by andrii-bodnar** [comment]
It looks like it's not that critical and a big codebase change won't be worth it, what do you think?

**Comment by philipp-winterle** [comment]
In the latest version within an react router 7 project I experience that` lignui extract` does hang endlessly when having the following code: 

```typescript
<Trans id="test">Message</Trans>
<Trans id="test">Message 2</Trans>
```

## Current Behaviour

extract runs endlessly

## Expected Behaviour

Error message "duplicate id with different message"

## Context

Iam using Trans macro to extract messages but I want to reuse them. I want to merge these ids with remote controlled ids which are managed by an admin interface. So this is the reason I need fixed ids. 

THe following things are hard to handle: 
- Same ids need to have the same message which is hard to maintain over a big codebase. Are there best practices Iam missing?
- Empty message is not usable

**Comment by ptmkenny** [comment]
I ran into the same problem as @philipp-winterle with a `t()` having duplicate IDs. There's no warning that the IDs are duplicated; instead, `extract` just hangs.  I'm running the latest version of lingui in React 19.1.
